### ISSUE TITLE
manual/hacking/Documentation.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#926]
- https://github.com/playframework-ja/translation-project/tree/2.3.5/manual/hacking/Documentation.md
- diffs between [2.2.0/manual/hacking/Documentation.md](https://github.com/playframework-ja/translation-project/tree/2.2.0/manual/hacking/Documentation.md) and [2.3.5/manual/hacking/Documentation.md](https://github.com/playframework-ja/translation-project/tree/2.3.5/manual/hacking/Documentation.md).

```
--- /Users/garbagetown/Desktop/2.2.0/manual/hacking/Documentation.md    2013-09-19 10:11:22.000000000 +0900
+++ //Users/garbagetown/Desktop/2.3.5/manual/hacking/Documentation.md   2014-07-28 12:26:38.000000000 +0900
@@ -1,3 +1,4 @@
+<!--- Copyright (C) 2009-2013 Typesafe Inc. <http://www.typesafe.com> -->
 # Guidelines for writing Play documentation

 The Play documentation is written in Markdown format, with code samples extracted from compiled, run and tested source files.
@@ -34,7 +35,7 @@
     "do something" in {
       //#some-label
       val msg = Seq("Hello", "world").mkString(" ")
-      //#same-label
+      //#some-label
       msg must_== "Hello world"
     }
   }
@@ -50,7 +51,7 @@
 * `###replace: foo` - Replace the next line with `foo`.  You may optionally terminate this command with `###`
 * `###insert: foo` - Insert `foo` before the next line.  You may optionally terminate this command with `###`
 * `###skip` - Skip the current line
-* `###skip n` - Skip the next n lines
+* `###skip: n` - Skip the next n lines

 For example:

@@ -108,3 +109,18 @@
 To ensure that the code samples compile, run and tests pass, run `./build test`.

 To validate that the documentation is structurely sound, run `./build validate-docs`.  This checks that there are no broken wiki links, code references or resource links, ensures that all documentation markdown filenames are unique, and ensures that there are no orphaned pages.
+
+## Code samples from external Play modules
+
+To avoid circular dependencies, any documentation that documents a Play module that is not a core part of Play can't include its code samples along with the rest of the Play documentation.  To address this, the documentation for that module can place an entry into the `externalPlayModules` map in `project/Build.scala`, including all the extra settings (namely library dependencies) required to build the code snippets for that module.  For example:
+
+'''scala
+val externalPlayModules: Map[String, Seq[Setting[_]]] = Map(
+  "some-module" -> Seq(
+    libraryDependencies += "com.example" %% "some-play-module" % "1.2.3" % "test"
+  ),
+  ...
+)
+'''
+
+Now place all code snippets that use that module in `code-some-module`.  Now to run any SBT commands, ensuring that that module is included, run `./build -Dexternal.modules=some-module test`, or to run the tests for all modules, run `./build -Dexternal-modules=all test`.

```

**Comment by garbagetown** [comment]
:hand:

**Comment by garbagetown** [comment]
merged at #1118
